addons/mail/static/src/models/messaging_initializer/messaging_initializer.js

```diff
@@ -19,25 +19,31 @@ function factory(dependencies) {
         async start() {
             await this.async(() => this.env.session.is_bound);
 
-            this.env.models['mail.thread'].create({
+            const inbox = this.env.models['mail.thread'].create({
                 id: 'inbox',
                 isPinned: true,
                 model: 'mail.box',
                 name: this.env._t("Inbox"),
             });
-            this.env.models['mail.thread'].create({
+            const starred = this.env.models['mail.thread'].create({
                 id: 'starred',
                 isPinned: true,
                 model: 'mail.box',
                 name: this.env._t("Starred"),
             });
-            this.env.models['mail.thread'].create({
+            const history = this.env.models['mail.thread'].create({
                 id: 'history',
                 isPinned: true,
                 model: 'mail.box',
                 name: this.env._t("History"),
             });
 
+            this.messaging.update({
+                history: [['link', history]],
+                inbox: [['link', inbox]],
+                starred: [['link', starred]],
+            });
+
```

Comment:
create + link should be done in one step (either "update relation with insert" on messaging, or add "link messaging" on create mailboxes).